buildSrc/src/main/kotlin/pk-publish.gradle.kts

```diff
@@ -1,69 +1,89 @@
+            publications.configureEach {
+                if (this is MavenPublication) {
+                    val publicName = "${rootProject.name} ${name.capitalize()}"
+                    pom {
+                        name.set(publicName)
+                        description.set("Powerful Dependency Injector for Kotlin")
+                        url.set("https://github.com/corbella83/PopKorn")
+                        licenses {
+                            license {
+                                name.set("The Apache Software License, Version 2.0")
+                                url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+                                distribution.set("repo")
+                            }
+                        }
+                        developers {
+                            developer {
+                                id.set("corbella83")
+                                name.set("Pau Corbella")
+                            }
+                        }
+                        scm {
+                            connection.set("scm:git:git://github.com/corbella83/PopKorn.git")
+                            developerConnection.set("scm:git:ssh://git@github.com/corbella83/PopKorn.git")
+                            url.set("https://github.com/corbella83/PopKorn")
+                        }
                     }
                 }
-                developers {
-                    developer {
-                        id.set("corbella83")
-                        name.set("Pau Corbella")
+            }
+
+            repositories {
+                maven {
+                    val releasesRepoUrl = uri(nexusPro)
+                    val snapshotsRepoUrl = uri(nexusSnapshot)
+                    url = if (isReleaseVersion) releasesRepoUrl else snapshotsRepoUrl
+                    credentials {
+                        username = nexusUsername
+                        password = nexusPassword
                     }
                 }
-                scm {
-                    connection.set("scm:git:git://github.com/corbella83/PopKorn.git")
-                    developerConnection.set("scm:git:ssh://git@github.com/corbella83/PopKorn.git")
-                    url.set("https://github.com/corbella83/PopKorn")
-                }
             }
         }
-    }
 
-    repositories {
-        maven {
-            val releasesRepoUrl = uri(properties.getProperty("nexus.pro_url"))
-            val snapshotsRepoUrl = uri(properties.getProperty("nexus.snap_url"))
-            url = if (isReleaseVersion) releasesRepoUrl else snapshotsRepoUrl
-            credentials {
-                username = properties.getProperty("nexus.username")
-                password = properties.getProperty("nexus.password")
-            }
+        tasks.withType<Sign>().configureEach {
+            onlyIf { isReleaseVersion }
         }
-    }
 
+        signing {
+            sign(publishing.publications)
+        }
+    }
```

Comment:
Show a compilation warning if local.properties is not found or some parameters are missing. Otherwise I won't have any hint of what is wrong when publishing

Reply:
Instead of showing a warning if `local.properties` is missing, I am going to add to the `else` branch which checks if all properties exists. Because `local.properties` can exists but it can have no configuration for publishing on maven central.